### PR TITLE
Add initial travis-ci integration

### PR DESCRIPTION
**Pull request by mestery**
Also make installing Docker optional, as travis-ci can do that for
us so no need to have ci/prepare.sh do that as well.

Signed-off-by: Kyle Mestery mestery@mestery.com

### CODE REVIEW

**Comment by mestery** [comment]
@huikang This works now, see here [1] for build history. I assume I need to hook the ovn-scale-test repository into my travis setup for this to work on the broader system? Or do we need to integrate it with the upstream OVS one? I will kickstart discussions with Ben and Justin on that.

[1] https://travis-ci.org/mestery/ovn-scale-test/builds

**Comment by huikang** [comment]
Nice work! We are expecting this for a long time. I will review the PR and provide the feedback soon. Thanks.

**Comment by huikang** [comment]
Can we add `set -o xtrace` in the CI scripts (e.g., [prepare.sh](https://github.com/openvswitch/ovn-scale-test/blob/master/ci/prepare.sh#L1)) for better debugging purpose?

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    # list docker-engine versions
+    - apt-cache madison docker-engine
+
+    # upgrade docker-engine to specific version
+    - sudo apt-get -o Dpkg::Options::="--force-confnew" install -y docker-engine=${DOCKER_VERSION}
+
+    # reinstall docker-compose at specific version
+    - sudo rm /usr/local/bin/docker-compose
+    - curl -L https://github.com/docker/compose/releases/download/${DOCKER_COMPOSE_VERSION}/docker-compose-`uname -s`-`uname -m` > docker-compose
+    - chmod +x docker-compose
```
Comment (huikang) [inline .travis.yml]: Is docker-compose required for running the CI?

Comment (mestery) [inline .travis.yml]: No, I will remove this part.

**Comment by mestery** [comment]
I have addred xtrace to all scripts.

**Comment by mestery** [comment]
Build 13 was a success:

https://travis-ci.org/mestery/ovn-scale-test/builds/139257265

**Comment by l8huang** [comment]
You guys make a great progress!

**Comment by huikang** [comment]
@l8huang we are still investigating why `set -o trace` does not provide expected trace info in travis-ci.
But, yes we are making progress : )

**Comment by mestery** [comment]
@huikang The latest version now has correct xtrace syntax. Other than the URL pointing at my travis-ci account it is ready to merge. We could merge it as is and then when Ben gives us access to the OVS travis-ci just update the link at that point.

**Comment by mestery** [comment]
You can see the xtrace results in build 14:

https://travis-ci.org/mestery/ovn-scale-test/builds/139571913

**Pull request merged**